router/usecase/pools/routable_cw_orderbook_pool.go

```diff
@@ -155,21 +155,28 @@ func (r *routableOrderbookPoolImpl) SetTokenOutDenom(tokenOutDenom string) {
 
 // CalcSpotPrice implements domain.RoutablePool.
 func (r *routableOrderbookPoolImpl) CalcSpotPrice(ctx context.Context, baseDenom string, quoteDenom string) (osmomath.BigDec, error) {
-	// Get the expected order direction
-	direction, err := r.GetDirection(quoteDenom, baseDenom)
+	// Get the expected order directionIn
+	directionIn, err := r.GetDirection(quoteDenom, baseDenom)
 	if err != nil {
 		return osmomath.BigDec{}, err
 	}
 
-	tickIdx, err := r.GetStartTickIndex(direction.Opposite())
+	directionOut := directionIn.Opposite()
+
+	tickIdx, err := r.GetStartTickIndex(directionOut)
 	if err != nil {
 		return osmomath.BigDec{}, err
 	}
 
 	tick := r.OrderbookData.Ticks[tickIdx]
 
 	// Calculate the price for the current tick
-	return clmath.TickToPrice(tick.TickId)
+	tickPrice, err := clmath.TickToPrice(tick.TickId)
+	if err != nil {
+		return osmomath.BigDec{}, err
+	}
+
+	return convertValue(osmomath.OneBigDec(), tickPrice, directionOut), nil
```

Comment:
Can be precomputed since `converValue` does not use mutative math

Reply:
I'm not sure what you mean. Could you elaborate?

Reply:
Define `var oneDec = osmomath.OneBigDec()` in global space and reference here